plugins/node/opentelemetry-instrumentation-mysql/src/instrumentation.ts

```diff
@@ -343,4 +384,56 @@ export class MySQLInstrumentation extends InstrumentationBase<
       };
     };
   }
+  private _setPoolcallbacks(
+    pool: mysqlTypes.Pool,
+    thisPlugin: MySQLInstrumentation,
+    id: string
+  ) {
+    //TODO:: use semantic convention
+    const poolName = id || getPoolName(pool);
+
+    pool.on('connection', connection => {
+      thisPlugin._connectionsUsage.add(1, {
+        state: 'idle',
+        name: poolName,
+      });
+      connection.on('end', () => {
+        thisPlugin._connectionsUsage.add(-1, {
+          state: 'idle',
+          name: poolName,
+        });
+      });
```

Comment:
Are we sure only an idle connection can end? Is it not possible for this to be a used connection?

Reply:
Oh, you're right! I tried to do: `pool.getConnection` > `conn.query` > `pool.end` - meaning without calling `conn.release` before calling `end`, and the result was adding (-1) to idle.... while it should be added (-1) to `used` (which never happened because user didn't call `release` and add (-1) to `idle`. right?

Reply:
Does the callback receive the connection? Is there a way to tell if it is idle?

Reply:
In [Pool.js](https://github.com/mysqljs/mysql/blob/master/lib/Pool.js#L16) there are arrays containing the connections. 
If I call `pool.getConnection` > `conn.query` > `conn.release` > `pool.end` - then the `end` function will remove the connection from `_freeConnections[]` and from `_allConnections[]` arrays, 
but If I call `pool.getConnection` > `conn.query` > `pool.end` without calling `conn.release` in the middle, then the `end` function will remove the connection only from `_allConnections[]` array,

Is it make sense to patch the `end` function and make our `end` be called before their `end`? maybe this way I can observe the pool on which `end` was called, and see if the `_freeConnections[]` contains the connection or not. 

of course I have an assumption here, that if a connection is in `_freeConnections[]` then `release` was already called and we already added (-1) to `used` and (+1) to `idle`.

Reply:
That might be the only way to do it if the callback for `end` doesn't receive the connection itself to inspect